core/src/main/scala/org/apache/spark/memory/StaticMemoryManager.scala

```diff
@@ -72,8 +72,11 @@ private[spark] class StaticMemoryManager(
       numBytes: Long,
       evictedBlocks: mutable.Buffer[(BlockId, BlockStatus)]): Boolean = synchronized {
     val currentUnrollMemory = storageMemoryPool.memoryStore.currentUnrollMemory
-    val maxNumBytesToFree = math.max(0, maxMemoryToEvictForUnroll - currentUnrollMemory)
-    val numBytesToFree = math.min(numBytes, maxNumBytesToFree)
+    val freeMemory = storageMemoryPool.memoryFree
+    val maxNumBytesToFree = math.max(0,
+      maxMemoryToEvictForUnroll - currentUnrollMemory - freeMemory)
```

Comment:
It's because we changed the semantics of `ensureFreeSpace` here. Previously, we want to ensure that this amount of total free space is available. Now, in `evictBlocksToFreeSpace`, we want to ensure an *extra* amount of free space is available. This means if we did not subtract `freeMemory` we would be double counting.